docs/r2/tensorboard_profiling_keras.ipynb

```diff
@@ -0,0 +1,1086 @@
+      ]
+    },
+    {
+      "metadata": {
+        "id": "djUvWu41mtXa",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "##### Copyright 2019 The TensorFlow Authors."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "su2RaORHpReL",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "#@title Licensed under the Apache License, Version 2.0 (the \"License\");\n",
+        "# you may not use this file except in compliance with the License.\n",
+        "# You may obtain a copy of the License at\n",
+        "#\n",
+        "# https://www.apache.org/licenses/LICENSE-2.0\n",
+        "#\n",
+        "# Unless required by applicable law or agreed to in writing, software\n",
+        "# distributed under the License is distributed on an \"AS IS\" BASIS,\n",
+        "# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
+        "# See the License for the specific language governing permissions and\n",
+        "# limitations under the License."
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "NztQK2uFpXT-",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "# TensorBoard Profile: Profiling basic training metrics in Keras\n",
+        "\n",
+        "<table class=\"tfo-notebook-buttons\" align=\"left\">\n",
+        "  <td>\n",
+        "    <img src=\"https://www.tensorflow.org/images/tf_logo_32px.png\" />Currently N/A\n",
+        "  </td>\n",
+        "  <td>\n",
+        "    <a target=\"_blank\" href=\"https://colab.research.google.com/github/tensorflow/tensorboard/blob/master/docs/r2/tensorboard_profiling_keras.ipynb\"><img src=\"https://www.tensorflow.org/images/colab_logo_32px.png\" />Run in Google Colab</a>\n",
+        "  </td>\n",
+        "  <td>\n",
+        "    <a target=\"_blank\" href=\"https://github.com/tensorflow/tensorboard/blob/master/docs/r2/tensorboard_profiling_keras.ipynb\"><img src=\"https://www.tensorflow.org/images/GitHub-Mark-32px.png\" />View source on GitHub</a>\n",
+        "  </td>\n",
+        "</table>"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "eDXRFe_qp5C3",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "\n",
+        "## Overview\n",
+        "Performance is critical for machine learning. TensorFlow has a built-in profiler that allows you to record runtime of each ops with very little effort. Then you can visualize the profile result in TensorBoard's **Profile Plugin**. This profiler only works for GPU and CPU workflow. For TPU users, please check out [Cloud TPU Tools](https://cloud.google.com/tpu/docs/cloud-tpu-tools) to learn how to profile on TPU.\n",
```

Comment:
The Cloud TPU guide also uses TensorBoard's profiler. Is the intention to say that this tutorial focuses only on GPU and CPU?

Reply:
Yes, they shared the same UI, but they have different setups

Reply:
It might be worth clarifying that the setup is different but both are visualized in the profiler. Maybe "this tutorial focuses on GPU but the profile dashboard can also be used with TPUs by following the guide here" (with a link).